profiling/src/string_table/bump_owned.rs

```diff
@@ -0,0 +1,70 @@
+use bumpalo::{collections, Bump};
+use std::ops::Range;
+
+pub struct BorrowedStringTable<'b> {
+    arena: &'b Bump,
+    pub set: super::borrowed::BorrowedStringTable<'b>,
+}
+
+impl<'b> BorrowedStringTable<'b> {
+    #[inline]
+    pub fn new(arena: &'b Bump) -> Self {
+        Self {
+            arena,
+            set: Default::default(),
+        }
+    }
+}
+
+impl<'b> super::StringTable for BorrowedStringTable<'b> {
+    #[inline]
+    fn len(&self) -> usize {
+        self.set.len()
+    }
+
+    fn insert_full(&mut self, str: &str) -> (usize, bool) {
+        match self.set.map.get(str) {
+            None => {
+                let owned = collections::String::from_str_in(str, self.arena);
```

Comment:
@morrisonlevi just for my understanding: this is where we actually store the string in the arena?

Reply:
Yes, this is where the sting is actually being stored. Docs:
https://docs.rs/bumpalo/latest/bumpalo/collections/string/struct.String.html#method.from_str_in

> Construct a new String<'bump> from a string slice.

```rust
use bumpalo::{Bump, collections::String};

let b = Bump::new();

let s = String::from_str_in("hello", &b);
assert_eq!(s, "hello");
```